lib/board.rb

```diff
@@ -0,0 +1,47 @@
+class Board
+
+  attr_accessor :cells
+
+  def initialize
+    @cells = Array.new(9, " ")
+  end
+
+  def display
+    puts " #{@cells[0]} | #{@cells[1]} | #{@cells[2]} "
+    puts "-----------"
+    puts " #{@cells[3]} | #{@cells[4]} | #{@cells[5]} "
+    puts "-----------"
+    puts " #{@cells[6]} | #{@cells[7]} | #{@cells[8]} "
+  end
+
+  def position(input)
+    @cells[input.to_i - 1]
+  end
+
+  def full?
+    !@cells.include?(" ")
+  end
+
+  def taken?(position)
+    #binding.pry
+    !(self.position(position) == " ")
+  end
+
+  def valid_move?(position)
+    position.to_i.between?(1,9) && !self.taken?(position.to_i)
+  end
+
+  def turn_count
+    @cells.select {|c| c == "X" || c == "O"}.count
+  end
+
+  def update(position, player)
+    position_adj = (position.to_i - 1)
+    @cells[position_adj] = player.token
+  end
+
+  def reset!
+    @cells.clear
+    @cells = Array.new(9, " ")
+  end
```

Comment:
`@cells.clear` is not needed since `@cells = Array.new(9, " ")` reassigns the value of `@cells`